src/provider.c

```diff
@@ -139,6 +140,10 @@ static void p11prov_ctx_free(P11PROV_CTX *ctx)
         }
     }
 
+    if (ctx->pin) {
```

Comment:
no need for the if here, openssl handles NULL pointers correctly

Reply:
Yes, but it calls `strlen` which is not guaranteed to handle NULL.

Reply:
uh?

Reply:
oh, doh!